s2s.zig

```diff
@@ -221,8 +229,12 @@ fn recursiveDeserialize(stream: anytype, comptime T: type, allocator: ?std.mem.A
                     const slice = try allocator.?.alloc(ptr.child, length);
                     errdefer allocator.?.free(slice);
 
-                    for (slice) |*item| {
-                        try recursiveDeserialize(stream, ptr.child, allocator, item);
+                    if (ptr.child == u8) {
+                        _ = try stream.readAll(slice);
```

Comment:
this is wrong. if the stream ends too early, you have a half-filled slice. Please use `readNoEof` here.
Ignoring the return value is a code smell and it's wrong here, as the `readAll()` might read zero bytes.